src/renderer/components/ReactFlowBox.tsx

```diff
@@ -255,14 +255,18 @@ export const ReactFlowBox = memo(({ wrapperRef, nodeTypes, edgeTypes }: ReactFlo
                             },
                         };
 
-                        edgesToRemove.concat(
-                            edges.filter(
-                                (e) =>
-                                    e.source === node.id &&
-                                    nodes.find((n) => n.id === e.target)?.parentNode !==
-                                        iterInBounds.id
-                            )
-                        );
+                        edgesToRemove = [
+                            ...edgesToRemove,
+                            ...edges.filter((e) => {
+                                if (e.source === node.id) {
+                                    const target = nodes.find((n) => n.id === e.target);
+                                    if (target) {
+                                        return target.parentNode !== iterInBounds.id;
+                                    }
+                                }
+                                return false;
+                            }),
+                        ];
```

Comment:
While I commend your commitment to immutable data structures, using `push` would probably be easier.
```suggestion
                        edgesToRemove.push(
                            ...edges.filter((e) => {
                                if (e.source === node.id) {
                                    const target = nodes.find((n) => n.id === e.target);
                                    if (target) {
                                        return target.parentNode !== iterInBounds.id;
                                    }
                                }
                                return false;
                            }),
                        );
```